openshift/templates/nginx-runtime/Dockerfile

```diff
@@ -1,8 +1,34 @@
 FROM nginx:mainline
-COPY ./s2i/bin/ /usr/libexec/s2i/
-LABEL io.openshift.s2i.scripts-url=image:///usr/libexec/s2i
+
+ENV STI_SCRIPTS_PATH=/usr/libexec/s2i
+
+COPY ./s2i/bin/ $STI_SCRIPTS_PATH/
+LABEL io.openshift.s2i.scripts-url=image://$STI_SCRIPTS_PATH
 COPY nginx.conf.template /tmp/
 COPY default.conf /etc/nginx/conf.d/
-RUN chmod -R 0777 /tmp /var /run /etc /mnt /usr/libexec/s2i/
+RUN mkdir -p /etc/nginx/
+RUN echo "" > /etc/nginx/publicServerEnvironmentSettings.js
+RUN mkdir -p /tmp/app/dist/
+RUN ln -sf /etc/nginx/publicServerEnvironmentSettings.js /tmp/app/dist/publicServerEnvironmentSettings.js
+# changed /etc to /etc/nginx as error thrown from ibm run cloud on this line
+# /etc/nginx was seen to require chmod 777 much later in deployment
+# RUN chmod -R 0777 /tmp /var /etc/nginx /mnt $STI_SCRIPTS_PATH/
+# =================================================================================
+# Fix up permissions
+# ref: https://torstenwalter.de/openshift/nginx/2017/08/04/nginx-on-openshift.html
+# - S2I sripts must be executable
+# - Make sure nginx can read and write it's working directories.
+# - The container dynamically configures nginx on startup
+# - The application artifacts live in /tmp
+# ---------------------------------------------------------------------------------
+RUN chmod -R g+rwx $STI_SCRIPTS_PATH
+RUN chmod g+rw /var/cache/nginx \
+               /var/run \
+               /var/log/nginx \
+               /etc/nginx/publicServerEnvironmentSettings.js \
+               /etc/nginx/nginx.conf \
+               /tmp
+# =================================================================================
+RUN chmod 777 /run
```

Comment:
Have we checked that these changes aren't breaking? It looks like the right permissions from epic, but good to be sure. I remember on epic that this image doesn't trigger a rebuild unless you manually do it. When it finally rebuilt and didn't work we had forgotten about this change.

Reply:
Nope, I haven't checked yet. I just wanted to get some eyes on the template updates before I started breaking environments completely.

Reply:
haha okay well as long as we remember to rebuild this image after we merge this. The rest of this pr looks good to me, just thought I would make a note of this so we don't run into the same problem again down the line.